server/services/ticket.js

```diff
@@ -41,10 +42,35 @@ function getTicketsByDate(filterOpts) {
   return Ticket.getTicketsByDate(filterOpts);
 }
 
+function addAction(ticketId, actionType, userId, device) {
+  const schema = {
+    ticketId: Joi.string().hex().length(24).required(),
+    actionType: Joi.any().only(_.keys(actionTypes)).required(),
+    userId: Joi.string().hex().length(24),
+    device: Joi.object()
+  };
+  return JoiHelper.validate({ ticketId, actionType, userId, device }, schema)
+    .then(values =>
+      Ticket.addAction(values.ticketId, values.actionType, values.userId, values.device));
+}
+
+function removeAction(ticketId, actionType, timestamp, userId) {
+  const schema = {
```

Comment:
Currently the action is removed by timestamp, actionType, and userId because those are always on the ticket. Thinking about it now, it makes more sense to add a UUID onto the actions. This also isn't actually on this PR, it's just moved from a different file, this change should be in a different PR.